#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 81a8df7a16adacb5a6c2a73b4a5bd75676af88a7 Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Tue, 30 Sep 2025 06:50:19 +0200
Subject: [PATCH] Fix bug 510055: When we press enter in search line we don't
 want to close dialog

By default sometime we press enter => it will close it's not that we
want

BUG: 510055
FIXED-IN: 6.19.0
---
 src/kpageview.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/kpageview.cpp b/src/kpageview.cpp
index b3b29d40..9c45d6bc 100644
--- a/src/kpageview.cpp
+++ b/src/kpageview.cpp
@@ -18,6 +18,7 @@

 #include <ktitlewidget.h>

+#include <KLineEditEventHandler>
 #include <QAbstractButton>
 #include <QAbstractItemView>
 #include <QApplication>
@@ -502,6 +503,7 @@ void KPageViewPrivate::init()
     layout->setRowStretch(3, 1);

     searchLineEdit = new QLineEdit(defaultWidget);
+    KLineEditEventHandler::catchReturnKey(searchLineEdit);
     searchTimer.setInterval(400);
     searchTimer.setSingleShot(true);
     searchTimer.callOnTimeout(q, [this] {
--
2.51.1
